examples/update_bids_datasets.py

```diff
@@ -0,0 +1,150 @@
+#
+# Download the data if it hasn't been downloaded already, and return the path
+# to the download directory. This directory is the so-called `root` of this
+# BIDS dataset.
+bids_root = somato.data_path()
+
+###############################################################################
+# Explore the dataset contents
+# ----------------------------
+#
+# We can use MNE-BIDS to print a tree of all
+# included files and folders. We pass the ``max_depth`` parameter to
+# `mne_bids.print_dir_tree` to the output to three levels of folders, for
+# better readability in this example.
+
+print_dir_tree(bids_root, max_depth=3)
+
+# We can generate a report of the existing dataset
+print(make_report(bids_root))
+
+###############################################################################
+# Update the sidecar JSON dataset contents
+# ----------------------------------------
+#
+# We can use MNE-BIDS to update all sidecar files for a matching
+# ``BIDSPath`` object. We then pass in a sidecar template to update
+# all matching metadata fields within the BIDS dataset.
+
+# create a BIDSPath object
+bids_root = somato.data_path()
+datatype = 'meg'
+subject = '01'
+task = 'somato'
+suffix = 'meg'
+
+bids_path = BIDSPath(subject=subject, task=task, suffix=suffix,
+                     datatype=datatype, root=bids_root)
+sidecar_path = bids_path.copy().update(extension='.json')
+
+# We can now retrieve a list of all MEG-related files in the dataset:
+print(bids_path.match())
+
+# Define a sidecar template as a dictionary
+entries = {
+    'PowerLineFrequency': 60,
+    'Manufacturer': 'Captrak',
+    'InstitutionName': 'Martinos Center'
+}
+
+# Note: ``update_sidecar_json`` will perform essentially a
+# dictionary update to your sidecar file, so be absolutely sure
+# that the ``entries`` are defined properly.
+#
+# Now update all sidecar fields according to the template
+update_sidecar_json(bids_path=sidecar_path, entries=entries)
+
+###############################################################################
+# Read the updated dataset
+# ------------------------
+#
+# We can use MNE-BIDS to update all sidecar files for a matching
+# ``BIDSPath`` object. We then pass in a sidecar template to update
+# all matching metadata fields within the BIDS dataset.
+
+# new line frequency is now 60 Hz
+raw = read_raw_bids(bids_path=bids_path)
+print(raw.info['line_freq'])
+
+###############################################################################
+# Generate a new report based on the updated metadata.
+
+# The manufacturer was changed to ``Captrak``
+print(make_report(bids_root))
+
+###############################################################################
+# We can also update sidecars by passing in a template JSON file.
+
+# update the template to have a new PowerLineFrequency
+entries['Manufacturer'] = 'Elekta'
+entries['PowerLineFrequency'] = 50
+
+# create a temporary JSON file that contains the updated template
```

Comment:
why do you need to do this? For an entry level example this looks like complex code. Can we avoid this?

Reply:
If I'm not misunderstanding anything,`entries` can be a dict, so I agree we should pass a dict here. Passing a JSON file is advanced usage that maybe can be mentioned, but this example here is a bit too complicated